airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -170,6 +172,7 @@ def __init__(self,  # pylint: disable=too-many-arguments,too-many-locals
                  schedulername: Optional[str] = None,
                  full_pod_spec: Optional[k8s.V1Pod] = None,
                  init_containers: Optional[List[k8s.V1Container]] = None,
+                 log_events_on_failure: bool = True,
```

Comment:
I think for backwards compatibility we could set it to False, though i am on the fence on this one. It's super-useful and it does not change the behaviour of the pod launcher but it prints useful log files on failure, so I am not sure if backwards compatibility is needed.

Reply:
That's a good point - It can result in an error if the RBAC policy isn't set to allow the airflow SA  `pods/get`

Reply:
Default set to False now.